### PR TITLE
Refactor:Remove GlobalC::ucell in the module_surchem

### PR DESCRIPTION
**Pull request by A-006**
### Unit Tests and/or Case Tests for my changes
- Add ucell in the module_surchem test.

### What's changed?
- Change the ucell in the module_surchem.
- Add const ucell for the strcture_factor function.

### CODE REVIEW

File: source/module_hamilt_general/module_surchem/cal_pseudo.cpp [hunk 1]
```diff
@@ -8,7 +8,8 @@ void surchem::gauss_charge(const UnitCell& cell,
                            complex<double>* N,
                            Structure_Factor* sf)
 {
-    sf->setup_structure_factor(&GlobalC::ucell, rho_basis); // call strucFac(ntype,ngmc)
+    UnitCell cell_tmp = cell;
```
Comment (mohanchen) [inline source/module_hamilt_general/module_surchem/cal_pseudo.cpp]: use pointer

File: source/module_hamilt_general/module_surchem/cal_pseudo.cpp [hunk 2]
```diff
@@ -8,7 +8,8 @@ void surchem::gauss_charge(const UnitCell& cell,
                            complex<double>* N,
                            Structure_Factor* sf)
 {
-    sf->setup_structure_factor(&GlobalC::ucell, rho_basis); // call strucFac(ntype,ngmc)
+    UnitCell* cell_tmp = const_cast<UnitCell*>(&cell);
```
Comment (mohanchen) [inline source/module_hamilt_general/module_surchem/cal_pseudo.cpp]: why use const_cast?

Comment (A-006) [reply source/module_hamilt_general/module_surchem/cal_pseudo.cpp]: This problem has been finished.

File: source/module_hamilt_general/module_surchem/cal_pseudo.cpp (hunk 1 above)
Comment (A-006) [reply source/module_hamilt_general/module_surchem/cal_pseudo.cpp]: The ucell has been modified back.

**Review by mohanchen — APPROVED** [review]

**Pull request merged**